src/ts/app/editor/src/render-run-button.ts

```diff
@@ -1,6 +1,26 @@
 import {html, TemplateResult} from "lit-html"
+import {Dispatch} from "redux"
+import {DispatchedAction} from "./action"
+import {programRunAction} from "app/program-run"
+import {model} from "app/domain"
 
 
-export function renderRunButton(userCanStartCodeRun: boolean): TemplateResult {
-    return html`<button class="run-button" ?disabled="${(!userCanStartCodeRun)}">Run</button>`
+export function renderRunButton(userCanStartCodeRun: boolean,
+                                currentPythonModule: model.PythonModule,
+                                dispatch: Dispatch<DispatchedAction>): TemplateResult {
+
+    function dispatchKickoffRun() {
+        if (userCanStartCodeRun) {
+            dispatch({
+                type: programRunAction.Keys.KICKOFF_RUN,
+                pythonModule: currentPythonModule
+            })
+        }
+    }
+
+    return html`
+        <button 
+            class="run-button" 
+            ?disabled="${(!userCanStartCodeRun)}"
```

Comment:
hmmm, maybe this is new to me, what does `?disabled` do? specifically the ?

Reply:
See "bind to a boolean attribute" here https://lit-element.polymer-project.org/guide/templates

Obscure but occasionally crucial feature.